src/transformers/modeling_utils.py

```diff
@@ -1569,6 +1581,82 @@ def retrieve_modules_from_names(self, names, add_prefix=False, remove_prefix=Fal
 
         return retrieved_modules
 
+    @classmethod
+    def _load_state_dict_into_model_low_mem_p1(cls, state_dict):
+        """
+        This is an experimental 2-part function that loads the model using ~1.x model size CPU memory
+
+        Part 1:
+
+        1. save which state_dict keys we have
+        2. drop state_dict
+        """
+        require_version_core("torch>=1.9")
+        if is_deepspeed_zero3_enabled():
+            raise ValueError("low_cpu_mem_usage arg cannot be used with DeepSpeed ZeRO-3")
```

Comment:
In the spirit of having this method be potentially reusable outside of the low-memory scope, these two commands aren't necessary (but they are necessary for the second part!)